Python/import.c

```diff
@@ -1666,10 +1666,42 @@ PyImport_ImportModuleLevelObject(PyObject *name, PyObject *globals,
         }
     }
     else {
+        static int import_level;
+        _Py_IDENTIFIER(importprofile);
+        int import_profile = 0;
+        _PyTime_t t1=0, t2;
+
         Py_XDECREF(mod);
+
+        /* XOptions is initialized after first some imports.
+         * So we can't have negative cache.
+         * Anyway, importlib.__find_and_load is much slower than
+         * _PyDict_GetItemId()
+         */
+        PyObject *xoptions = PySys_GetXOptions();
+        if (xoptions) {
+            PyObject *value = _PyDict_GetItemId(xoptions, &PyId_importprofile);
+            import_profile = (value == Py_True);
+        }
+
+        if (import_profile) {
+            import_level++;
+            t1 = _PyTime_GetMonotonicClock();
+        }
+
         mod = _PyObject_CallMethodIdObjArgs(interp->importlib,
                                             &PyId__find_and_load, abs_name,
                                             interp->import_func, NULL);
+
+        if (import_profile) {
+            import_level--;
+            t2 = _PyTime_GetMonotonicClock();
+            fprintf(stderr, "%*s- %s %ld [us]\n",
+                    import_level*2, "",
+                    PyUnicode_AsUTF8(abs_name),
+                    (long)(t2 - t1) / 1000);
```

Comment:
Would it be possible to use the ms unit if us >= 10 000? It would make the output easier to read, no?

I don't understand the "123 [us]" format. Why not "123 us"?

Reply:
I would vote for "123 us" too, or "[123 us]".

Reply:
Square brackets is used to make it clear it's not variable (123 * u * s), but physical unit (123 in μs).
In English, it's rarely used, only when it can be confusing.

In Japan, some community use "use square bracket always for physical unit" habit.
And I studied in such college.
I just know it's local habit and abuse in general!

Reply:
> Would it be possible to use the ms unit if us >= 10 000? It would make the output easier to read, no?

I prefer fixed unit.  It makes post analysis (e.g. generate [Flame Graph](http://www.brendangregg.com/FlameGraphs/cpuflamegraphs.html)) easy.

Reply:
Agreed with @methane. Mixed units are confusing when values are meant to be compared.